### PR TITLE
Expose list of projects in cache to Python API

### PR DESCRIPTION
**Pull request by pblottiere**
## Description

``` python
$ print(QgsConfigCache.instance().projects())
[<QgsProject: 'QGIS-Training-Data/exercise_data/qgis-server-tutorial-data/world.qgs'>, <QgsProject: 'postgresql:?service=myservice&schema=public&project=myproject'>]
```

### CODE REVIEW

**Comment by elpaso** [comment]
Nice! Can you add a test case for the new method?

File: python/server/auto_generated/qgsconfigcache.sip.in [hunk 1]
```diff
@@ -66,6 +66,13 @@ value is ``False``).
 Returns the name of the current strategy
 
 .. versionadded:: 3.26
+%End
+
+    QMap<QString, QDateTime> projects() const;
+%Docstring
+Returns projects currently in cache.
```
Comment (m-kuhn) [inline python/server/auto_generated/qgsconfigcache.sip.in:73]: Can we be more precise _what_ we will get?
Intuitively I was assuming a list of project objects from given `projects()` but your example in the PR made it clearer. I think the API docs should be precise on what we can expect from the string and the date.

**Comment by rldhont** [comment]
@dmarteau ping

**Comment by dmarteau** [comment]
@pblottiere Why not returning the list of QgsProject instead only names ?  That could be useful for introspection without the need to load the project a second time ? (especially if the project as a huge memory payload)

**Comment by pblottiere** [comment]
Hello @dmarteau,

> Why not returning the list of QgsProject instead only names ? That could be
> useful for introspection without the need to load the project a second time ?
> (especially if the project as a huge memory footprint)

Actually you can use the path to retrieve the project in cache with
`QgsConfigCache.instance().project(path)` (so we don't load the project a second time).

But anyway, I don't have any objections if you think it's more convenient. I'm going to take a look.

**Comment by m-kuhn** [comment]
Sounds good to me, one less lookup for all information

**Comment by pblottiere** [comment]
Hello @elpaso @m-kuhn and @dmarteau,

I took into account your comments, let me know if something still needs to be updated.

Thanks for your review :).

**Comment by m-kuhn** [comment]
Thanks @pblottiere did you have a chance to review the api docs too?

**Comment by pblottiere** [comment]
Hello @m-kuhn @elpaso @rldhont,

I think this PR is ready (doc is ok, unit tests have been added, CI is green and I just rebased).  

I just need a final review to merge this PR :).

Thanks!

**Review by rldhont — APPROVED** [review]

**Comment by pblottiere** [comment]
Thanks @rldhont :)

**Pull request merged**